### PR TITLE
M3: Remove deprecations from SmsBundle

### PR DESCRIPTION
**Pull request by hluchas**
**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/8006
| BC breaks? | Y
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Remove deprecated code from SmsBundle

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com/)
2. Try to configure SMS transport with Twilio
3. Try to use SMS form to edit/create
4. Try to configure SMS send via campaign
5. Try to send SMS via campaign

#### List backwards compatibility breaks:
1. \Mautic\SmsBundle\Form\Type\SmsType::__constructor() removed
2. Remove unused Mautic\SmsBundle\Event\SmsClickEvent
3. Remove unused \Mautic\SmsBundle\Exception\MissingPasswordException
4. Remove unused \Mautic\SmsBundle\Exception\MissingUsernameException

### CODE REVIEW

File: app/bundles/SmsBundle/Views/FormTheme/SmsSendList/smssend_list_row.html.php [hunk 1]
```diff
@@ -8,11 +8,16 @@
...
+use Mautic\SmsBundle\Form\Type\SmsType;
+
 ?>
 
 <div class="row">
     <div class="col-xs-8">
-        <?php echo $view['form']->row($form['sms']); ?>
+        <?php
+
+        echo $view['form']->row($form[SmsType::class]); ?>
```
Comment (escopecz) [inline app/bundles/SmsBundle/Views/FormTheme/SmsSendList/smssend_list_row.html.php]: Remove the empty lines above to make it one-liner again.

**Comment by escopecz** [comment]
@hluchas please resolve the conflicts and fix the description.

**Comment by hluchas** [comment]
> @hluchas please resolve the conflicts and fix the description.

Fixed

File: app/bundles/SmsBundle/Helper/SmsHelper.php [hunk 2]
```diff
@@ -59,25 +61,28 @@ class SmsHelper
...
+     * @param DoNotContact      $doNotContact
      */
-    public function __construct(EntityManager $em, LeadModel $leadModel, PhoneNumberHelper $phoneNumberHelper, SmsModel $smsModel, IntegrationHelper $integrationHelper)
-    {
+    public function __construct(EntityManager $em,
+        LeadModel $leadModel,
+        PhoneNumberHelper $phoneNumberHelper,
+        SmsModel $smsModel,
+        IntegrationHelper $integrationHelper,
+        DoNotContact $doNotContact
```
Comment (escopecz) [inline app/bundles/SmsBundle/Helper/SmsHelper.php]: When you go to the Mautic config page you get:
```
Error: Uncaught Symfony\Component\Debug\Exception\FatalThrowableError: Type error: Too few arguments to function Mautic\SmsBundle\Helper\SmsHelper::__construct(),
```

**Comment by escopecz** [comment]
When trying to load the Text message campaign action form:
Screenshot 2019-11-28 at 11 00 30

**Pull request merged**